Simplify authoring guidelines for "Reference Solution" snippet (#179)
codewars/docs: content/authoring/guidelines/reference-solution.md

# Writing a Reference Solution

> The reference solution snippet, created by the kata author or translator, is a snippet which might appear as not very relevant, but when prepared incorrectly, it still can incur some additional, unnecessary maintenance cost. It might appear that reference solution is just a solution like every other submitted by a user, and the only important thing is to pass the tests. However, it's something more: reference solution is run every time a kata or translation is republished, and it should be prepared in a way which would prevent the publication from failing in an unexpected way.

Reviewer: ```suggestion
The reference solution snippet, created by the kata author or translator, is a snippet which might appear as not very relevant, but when prepared incorrectly, it still can incur some additional, unnecessary maintenance cost. It might appear that the reference solution is just a solution like every other submitted by a user, and that the only important thing is to pass the tests. However, it's something more: the reference solution is run every time a kata or translation is republished, and it should be prepared in a way which would prevent the publication from failing in an unexpected way.
```

> For above reasons, this snippet **should conform to [General Coding Guidelines][authoring-guidelines-general-coding]**: the reference solution is code too, and, as such, should keep up to code quality standards. It will make maintenance of the kata much easier, and the readability of the reference solution can be very helpful for future maintainers and translators.

Reviewer: ```suggestion
For the above reasons, this snippet **should conform to [General Coding Guidelines][authoring-guidelines-general-coding]**: the reference solution is code too, and, as such, should keep up to code quality standards. It will make maintenance of the kata much easier, and the readability of the reference solution can be very helpful for future maintainers and translators.
```

> The reference solution snippet, created by the kata author or translator, is a snippet which might appear as not very relevant, but when prepared incorrectly, it still can incur some additional, unnecessary maintenance cost. It might appear that the reference solution is just a solution like every other submitted by a user, and that the only important thing is to pass the tests. However, it's something more: the reference solution is run every time a kata or translation is republished, and it should be prepared in a way which would prevent the publication from failing in an unexpected way.
> 
> For the above reasons, this snippet **should conform to [General Coding Guidelines][authoring-guidelines-general-coding]**: the reference solution is code too, and, as such, should keep up to code quality standards. It will make maintenance of the kata much easier, and the readability of the reference solution can be very helpful for future maintainers and translators.

Reviewer 2: Isn't this too wordy? It's difficult to understand because of that and the passive tone.

```suggestion
The reference solution **should conform to [General Coding Guidelines][authoring-guidelines-general-coding]** because it must be maintained, unlike other submitted solutions. Always think about the future maintainers and translators.
```

I'm not sure having a separate page for this is worth it, but I do hate fixing **inelegant** one-liners that the translator was once proud of before disappearing.

Reviewer 3: (note: the smiley is about the comment below the suggestion)

What is the "reference solution" should be told first, I believe. Currently, this cannot be deduced from the text. For instance, when I hear "ref solutino", I always think about the one used for the random tests, in the tests suite.
Using kazk suggestion:

```suggestion
The reference solution is the solution of the author/translator that is showing up in the kata solutions page. It **should conform to [General Coding Guidelines][authoring-guidelines-general-coding]** because it must be maintained if changes are done to the tests, unlike other submitted solutions. Always think about the future maintainers and translators.
```

Author: I wanted users to realize that this snippet poses a maintenance cost, and not that it's just "write once and forget". I have an impression that users do not realize that sloppiness here can cause some unnecessary struggle to maintainers. But indeed I could be too wordy, because I often am :)
